File: answers/C/@mykeels/17-convert-decimal-to-base-n.c
#include <stdio.h>
#include <assert.h>
#include <stdlib.h>
#include <string.h>

char* toBaseN(int decimalValue, int n);

int main() {
    while (1) {
        int n, value;

        printf("Enter the decimal value you wish to convert: ");
        scanf("%d", &value);
        
        printf("Enter the value of base n: ");
        scanf("%d", &n);

        printf("%d base 10 = %s base %d\n\n", value, toBaseN(value, n), n);
    }
    return 0;
}

char* toBaseN(int decimalValue, int n) {
    assert(n <= 10);
    
    char* sResult = malloc (sizeof (char) *128);

    int position = 0;
    int div = decimalValue;
    int rem = 0;

    while (div > 0) {
        div = div / n;
        rem = decimalValue % n;
        sResult[position] = rem + '0';
        position++;
        decimalValue = div;
    }
    
    char* result = malloc (sizeof (char) *(position + 1));

    for (int i = position - 1; i >= 0; i--) {
        result[position - 1 - i] = sResult[i];
    }
    result[position] = '\0';

    return result;
}

File: answers/Go/@codehakase/17-convert-decimal-to-base-n.go
package main

import (
	"fmt"
)

func main() {
	var n, value int
	fmt.Println("Enter the decimal value you wish to convert: ")
	fmt.Scanf("%d\n", &value)

	fmt.Println("Enter the value of base n: ")
	fmt.Scanf("%d\n", &n)

	fmt.Printf("%d base 10 = %d base %d\n\n", value, toBaseN(value, n), n)
}

func toBaseN(number, base int) int {
	var result int
	res := []int{}
	remainder := 0
	done := false
	for !done {
		if number == 0 {
			done = true
		} else {
			number = number / base
			remainder = number % base
			res = append(res, remainder)
		}
		remainder = 0
	}
	for i := len(res) - 1; i >= 0; i-- {
		result += res[i]
	}
	return result
}
